plugins/woocommerce/src/Blocks/Utils/ProductCollectionUtils.php

```diff
@@ -92,4 +99,171 @@ private static function remove_empty_array_recursive( $array ) {
 		}
 		return $array;
 	}
+
+	/**
+	 * Parse editor's location context from WP Post.
+	 *
+	 * Possible contexts:
+	 * - post
+	 * - page
+	 * - product
+	 * - product-archive
+	 * - cart
+	 * - checkout
+	 * - catalog
+	 * - order
+	 *
+	 * @param WP_Post $post The Post instance.
+	 *
+	 * @return string Returns the context.
+	 */
+	public static function parse_editor_location_context( $post ) {
+		$context = 'site';
+
+		if ( ! $post instanceof \WP_Post ) {
+			return $context;
+		}
+
+		$post_type = $post->post_type;
+		if ( ! in_array( $post_type, array( 'post', 'page', 'wp_template', 'wp_template_part' ), true ) ) {
+			return $context;
+		}
+
+		if ( in_array( $post_type, array( 'wp_template', 'wp_template_part' ), true ) ) {
+
+			$name = $post->post_name;
+			if ( false !== strpos( $name, SingleProductTemplate::SLUG ) ) {
+				$context = 'product';
+			} elseif ( ProductAttributeTemplate::SLUG === $name ) {
+				$context = 'product-archive';
+			} elseif ( false !== strpos( $name, 'taxonomy-' ) ) { // Including the '-' in the check to avoid false positives.
+				$taxonomy           = str_replace( 'taxonomy-', '', $name );
+				$product_taxonomies = get_object_taxonomies( 'product', 'names' );
+				if ( in_array( $taxonomy, $product_taxonomies, true ) ) {
+					$context = 'product-archive';
+				}
+			} elseif ( in_array( $name, array( CartTemplate::SLUG, MiniCartTemplate::SLUG ), true ) ) {
+				$context = 'cart';
+			} elseif ( CheckoutTemplate::SLUG === $name ) {
+				$context = 'checkout';
+			} elseif ( ProductCatalogTemplate::SLUG === $name ) {
+				$context = 'catalog';
+			} elseif ( OrderConfirmationTemplate::SLUG === $name ) {
+				$context = 'order';
+			}
+		}
+
+		if ( 'page' === $post_type ) {
+			$context = 'page';
+		}
+		if ( 'post' === $post_type ) {
+			$context = 'post';
+		}
+
+		return $context;
+	}
+
+	/**
+	 * Track usage of the Product Collection block within the given blocks.
+	 *
+	 * @param array $blocks     The parsed blocks to check.
+	 * @param bool  $in_single  Whether we are in a single product container (used for keeping state in the recurring process).
+	 *
+	 * @return array Parsed instances of the Product Collection block.
+	 */
+	public static function parse_blocks_track_data( $blocks, $in_single = false ) {
+
+		$instances = array();
+
+		foreach ( $blocks as $block ) {
+			if ( 'woocommerce/product-collection' === $block['blockName'] ) {
+
+				$instances[] = array(
+					'collection'        => $block['attrs']['collection'] ?? 'catalog',
```

Comment:
We refer to default one as `product-catalog`. I think it's good to keep it consistent

```suggestion
					'collection'        => $block['attrs']['collection'] ?? 'product-catalog',
```

Reply:
Done in https://github.com/woocommerce/woocommerce/pull/46466/commits/da2f7b41e564489fb2c59b1700b831da7f1117e2